src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -1039,6 +1040,24 @@ public static bool IsDefaultExpressionContext(this SyntaxTree syntaxTree, int po
                 return true;
             }
 
+            if (allowedModifiersBefore != null &&
```

Comment:
the cases enumerate all the things that will be checkedin this method.  You've adde da new thing to check for.